sdk/deepbook/src/client.ts

```diff
@@ -329,12 +377,12 @@ export class DeepBookClient {
 	 */
 	async swapExactQuoteForBase(
 		poolId: string,
-		tokenObjectIn: string,
-		amountIn: bigint,
+		tokenObjectIn: TransactionResult | TransactionObjectArgument | string,
```

Comment:
not a big fan of this, pretty confusing to see `tokenObjectIn` accepting a `TransactionResult` here

Reply:
they are basically outputs of `splitCoins`. We need to be able to pass in either a string ID or the whole coin.